### PR TITLE
feat(LiveQuery): Support $eq

### PR DESCRIPTION
**Pull request by sadakchap**
### New Pull Request Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues/7606).

### Issue Description

As per this [issue](https://github.com/parse-community/Parse-SDK-JS/issues/1372), currently it is not possible to combine `equalTo` clause with any other clauses and vice-versa using Parse-SDK-JS. After adding support for combining `equalTo` clause with other clauses in [Parse-SDK-JS's pr.](https://github.com/parse-community/Parse-SDK-JS/pull/1373)

We need to add support `equalTo` for LiveQuery, so that it triggers correct events for client.

Related issue: 7606

### Approach

add support for $eq query in [LiveQuery's matchKeyConstraints](https://github.com/parse-community/parse-server/blob/20cb3333aba67fb89e4e2cda586c559bdb4a23ef/src/LiveQuery/QueryTools.js#L148).

### TODOs before merging

- [ ] Add test cases
- [ ] Add entry to changelog
- [ ] Add changes to documentation (guides, repository pages, in-code descriptions)
- [ ] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [ ] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

**Comment by cbaker6** [comment]
Looks related to https://github.com/parse-community/parse-server/pull/7227#discussion_r611092794

**Comment by mtrezza** [comment]
Do you need a hand with this PR, or this ready for review?

**Comment by mtrezza** [comment]
@sadakchap Just a friendly ping, if you could add the tests that would be great, then we can merge and close the dashboard null filter issue as well. 🙂

**Comment by mtrezza** [comment]
@sadakchap please let me know when this is ready for review.

**Comment by mtrezza** [comment]
@sadakchap is this PR ready for review?

**Comment by sadakchap** [comment]
> @sadakchap is this PR ready for review?

Yes @mtrezza, I believe it is ready to be reviewed now.

File: spec/QueryTools.spec.js [hunk 1]
```diff
@@ -255,6 +255,71 @@ describe('matchesQuery', function () {
     expect(matchesQuery(img, q)).toBe(false);
   });
 
+  it('matches on queries with new format #parse-SDK-JS/pull/1373', function () {
```
Comment (mtrezza) [inline spec/QueryTools.spec.js:258]: How is this related to "add support for $eq in LiveQuery"? This is not a LiveQuery test. Is this a test for https://github.com/parse-community/Parse-SDK-JS/pull/1373 which is a separate issue?

Comment (sadakchap) [reply spec/QueryTools.spec.js:258]: Since in https://github.com/parse-community/Parse-SDK-JS/pull/1373, we are changing the format of `query`. There are certain LiveQuery tests(parse-sdk-js) which are failing because parse-server doesn't support new query format for LiveQuery.

So, we added support for new `query` format in LiveQuery in parse-server.

Comment (mtrezza) [reply spec/QueryTools.spec.js:258]: Not sure I follow; is https://github.com/parse-community/Parse-SDK-JS/pull/1373 a requirement for this PR? I don't think so, otherwise the tests in this PR wouldn't already pass.

Comment (sadakchap) [reply spec/QueryTools.spec.js:258]: Actually its the other, please follow this https://github.com/parse-community/Parse-SDK-JS/pull/1373#issuecomment-934451402

Comment (mtrezza) [reply spec/QueryTools.spec.js:258]: So I understand that the tests in https://github.com/parse-community/Parse-SDK-JS/pull/1373 will only pass when this PR (#7607) has been merged. But they relate to 2 different features:
- add support for $eq in LiveQuery
- add support to "combine equalTo clauses with [other] clauses"

So why is https://github.com/parse-community/Parse-SDK-JS/pull/1373 mentioned in this test case if this PR (#7607) has no dependency on https://github.com/parse-community/Parse-SDK-JS/pull/1373, only the other way around?

Comment (sadakchap) [reply spec/QueryTools.spec.js:258]: @mtrezza sorry, I could not understand you properly, could you please elaborate a bit more?

Comment (mtrezza) [reply spec/QueryTools.spec.js:258]: Sure, it seems these are 2 different features: 

- https://github.com/parse-community/parse-server/pull/7607
- https://github.com/parse-community/Parse-SDK-JS/pull/1373

Even if https://github.com/parse-community/Parse-SDK-JS/pull/1373 never gets merged, we would still want to merge https://github.com/parse-community/parse-server/pull/7607 as a feature, right? So I don't understand why the tests in https://github.com/parse-community/parse-server/pull/7607 mention https://github.com/parse-community/Parse-SDK-JS/pull/1373:
```
it('matches on queries with new format #parse-SDK-JS/pull/1373', function () {
```

**Comment by cbaker6** [comment]
@mtrezza i think you can close this via #8614

**Comment by mtrezza** [comment]
Closing via https://github.com/parse-community/parse-server/pull/8614
